tests/performance_tests/ChainPluginArgs.py

```diff
@@ -0,0 +1,207 @@
+    transactionFinalityStatusSuccessDurationSec: int=None
+    _transactionFinalityStatusSuccessDurationSecNodeosDefault: int=180
+    _transactionFinalityStatusSuccessDurationSecNodeosArg: str="--transaction-finality-status-success-duration-sec"
+    transactionFinalityStatusFailureDurationSec: int=None
+    _transactionFinalityStatusFailureDurationSecNodeosDefault: int=180
+    _transactionFinalityStatusFailureDurationSecNodeosArg: str="--transaction-finality-status-failure-duration-sec"
+    integrityHashOnStart: bool=None
+    _integrityHashOnStartNodeosDefault: bool=False
+    _integrityHashOnStartNodeosArg: str="--integrity-hash-on-start"
+    integrityHashOnStop: bool=None
+    _integrityHashOnStopNodeosDefault: bool=False
+    _integrityHashOnStopNodeosArg: str="--integrity-hash-on-stop"
+    blockLogRetainBlocks: int=None
+    _blockLogRetainBlocksNodeosDefault: int=False
+    _blockLogRetainBlocksNodeosArg: str="--block-log-retain-blocks"
+    genesisJson: str=None
+    _genesisJsonNodeosDefault: str=None
+    _genesisJsonNodeosArg: str="--genesis-json"
+    genesisTimestamp: str=None
+    _genesisTimestampNodeosDefault: str=None
+    _genesisTimestampNodeosArg: str="--genesis-timestamp"
+    printGenesisJson: bool=None
+    _printGenesisJsonNodeosDefault: bool=False
+    _printGenesisJsonNodeosArg: str="--print-genesis-json"
+    extractGenesisJson: bool=None
+    _extractGenesisJsonNodeosDefault: bool=False
+    _extractGenesisJsonNodeosArg: str="--extract-genesis-json"
+    printBuildInfo: bool=None
+    _printBuildInfoNodeosDefault: bool=False
+    _printBuildInfoNodeosArg: str="--print-build-info"
+    extractBuildInfo=None
+    _extractBuildInfoNodeosDefault=None
+    _extractBuildInfoNodeosArg: str="--extract-build-info"
+    forceAllChecks: bool=None
+    _forceAllChecksNodeosDefault: bool=False
+    _forceAllChecksNodeosArg: str="--force-all-checks"
+    disableReplayOpts: bool=None
+    _disableReplayOptsNodeosDefault: bool=False
+    _disableReplayOptsNodeosArg: str="--disable-replay-opts"
+    replayBlockchain: bool=None
+    _replayBlockchainNodeosDefault: bool=False
+    _replayBlockchainNodeosArg: str="--replay-blockchain"
+    hardReplayBlockchain: bool=None
+    _hardReplayBlockchainNodeosDefault: bool=False
+    _hardReplayBlockchainNodeosArg: str="--hard-replay-blockchain"
+    deleteAllBlocks: bool=None
+    _deleteAllBlocksNodeosDefault: bool=False
+    _deleteAllBlocksNodeosArg: str="--delete-all-blocks"
+    truncateAtBlock: int=None
+    _truncateAtBlockNodeosDefault: int=0
+    _truncateAtBlockNodeosArg: str="--truncate-at-block"
+    terminateAtBlock: int=None
+    _terminateAtBlockNodeosDefault: int=0
+    _terminateAtBlockNodeosArg: str="--terminate-at-block"
+    snapshot: str=None
+    _snapshotNodeosDefault: str=None
+    _snapshotNodeosArg: str="--snapshot"
+
+    def threads(self, threads: int):
+        self.chainThreads=threads
+
+    def supportedNodeosArgs(self) -> list:
```

Comment:
This and `__str__()` are wonderfully generic.  I think it's time to create a parent class for all the PluginArgs family, so there's just one copy of these two methods.

Reply:
Addressed: https://github.com/AntelopeIO/leap/pull/546/commits/0cd997ff6e547526afdf7379c2b911cdfd7d0113